flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartBackoffTimeStrategyFactoryLoader.java

```diff
@@ -0,0 +1,135 @@
+	 *     <li>Strategy set in the cluster(server-side) config (flink-conf.yaml),
+	 * unless the strategy is not specified</li>
+	 *     <li>{@link FixedDelayRestartBackoffTimeStrategy.FixedDelayRestartBackoffTimeStrategyFactory} if
+	 * checkpointing is enabled. Otherwise {@link NoRestartBackoffTimeStrategy.NoRestartBackoffTimeStrategyFactory}</li>
+	 * </ol>
+	 *
+	 * @param jobRestartStrategyConfiguration restart configuration given within the job graph
+	 * @param clusterConfiguration cluster(server-side) configuration
+	 * @param isCheckpointingEnabled if checkpointing is enabled for the job
+	 * @return new version restart strategy factory
+	 */
+	public static RestartBackoffTimeStrategy.Factory createRestartStrategyFactory(
+			final RestartStrategies.RestartStrategyConfiguration jobRestartStrategyConfiguration,
+			final Configuration clusterConfiguration,
+			final boolean isCheckpointingEnabled) {
+
+		checkNotNull(jobRestartStrategyConfiguration);
+		checkNotNull(clusterConfiguration);
+
+		return getJobRestartStrategyFactory(jobRestartStrategyConfiguration)
+			.orElse(getClusterRestartStrategyFactory(clusterConfiguration)
+				.orElse(getDefaultRestartStrategyFactory(isCheckpointingEnabled)));
+	}
+
+	private static Optional<RestartBackoffTimeStrategy.Factory> getJobRestartStrategyFactory(
+			final RestartStrategies.RestartStrategyConfiguration restartStrategyConfiguration) {
+
+		if (restartStrategyConfiguration instanceof RestartStrategies.NoRestartStrategyConfiguration) {
+			return Optional.of(NoRestartBackoffTimeStrategy.NoRestartBackoffTimeStrategyFactory.INSTANCE);
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FixedDelayRestartStrategyConfiguration) {
+			final RestartStrategies.FixedDelayRestartStrategyConfiguration fixedDelayConfig =
+				(RestartStrategies.FixedDelayRestartStrategyConfiguration) restartStrategyConfiguration;
+
+			return Optional.of(new FixedDelayRestartBackoffTimeStrategy.FixedDelayRestartBackoffTimeStrategyFactory(
+				fixedDelayConfig.getRestartAttempts(),
+				fixedDelayConfig.getDelayBetweenAttemptsInterval().toMilliseconds()));
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FailureRateRestartStrategyConfiguration) {
+			final RestartStrategies.FailureRateRestartStrategyConfiguration failureRateConfig =
+				(RestartStrategies.FailureRateRestartStrategyConfiguration) restartStrategyConfiguration;
+
+			return Optional.of(new FailureRateRestartBackoffTimeStrategy.FailureRateRestartBackoffTimeStrategyFactory(
+				failureRateConfig.getMaxFailureRate(),
+				failureRateConfig.getFailureInterval().toMilliseconds(),
+				failureRateConfig.getDelayBetweenAttemptsInterval().toMilliseconds()));
+		} else if (restartStrategyConfiguration instanceof RestartStrategies.FallbackRestartStrategyConfiguration) {
+			return Optional.empty();
+		} else {
+			throw new IllegalArgumentException("Unknown restart strategy configuration " +
```

Comment:
Branch is not tested but it's acceptable.

Reply:
Yes. I'd like to avoid adding a testing RestartStrategyConfiguration just for testing this branch.